lms/djangoapps/experiments/flags.py

```diff
@@ -155,23 +173,35 @@ def get_bucket(self, course_key=None, track=True):
             return self._cache_bucket(experiment_name, 0)
 
         # Check if the enrollment should even be considered (if it started before the experiment wants, we ignore)
-        if course_key and self.experiment_id is not None:
+        if self.experiment_id is not None:
             values = ExperimentKeyValue.objects.filter(experiment_id=self.experiment_id).values('key', 'value')
             values = {pair['key']: pair['value'] for pair in values}
 
             if not self._is_enrollment_inside_date_bounds(values, user, course_key):
                 return self._cache_bucket(experiment_name, 0)
 
-        bucket = stable_bucketing_hash_group(experiment_name, self.num_buckets, user.username)
-
-        # Now check if the user is forced into a particular bucket, using our subordinate bucket flags
+        # Determine the user's bucket.
+        # First check if forced into a particular bucket, using our subordinate bucket flags.
+        # If not, calculate their default bucket using a consistent hash function.
         for i, bucket_flag in enumerate(self.bucket_flags):
-            if bucket_flag.is_enabled(course_key):
+            forced_into_bucket = (
+                bucket_flag.is_enabled(course_key) if course_key
+                else bucket_flag.is_enabled_without_course_context()
+            )
+            if forced_into_bucket:
                 bucket = i
                 break
+        else:
+            bucket = stable_bucketing_hash_group(
+                bucketing_group_name, self.num_buckets, user.username
+            )
```

Comment:
(the only effective change here is that we now call `stable_bucketing_hash_group(...)` with `bucketing_group_name` instead of `experiment_name`)